### PR TITLE
Improved CONTRIBUTING Docs with 4+1 view and Overview Diagram

### PR DESCRIPTION
**Pull request by jalr4ever**
## Description

Update the developer contribution documentation and README.

## Motivation and Context

The current SDG has a good design and architecture, but it still poses some difficulty for most developers to get started. This PR adds more overview information about the design and functional descriptions, aiming to help developers better engage with SDG development.

## How has this been tested?

Document (doc) updated, no testing required.

## Types of changes

- [x] Maintenance (no change in code, maintain the project's CI, docs, etc.)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by jalr4ever** [comment]
@MooooCat Please help me review this PR.

**Comment by Wh1isper** [comment]
Thanks for providing such a great documentation improvement!

Notice that we still need to modify the docs here:  https://github.com/hitsz-ids/synthetic-data-generator/blob/ed60e49a1121faca05ae68f6ff2ea86f83a37b67/docs/source/developer_guides/contributing.rst, we need to either export as image or just offer a link to github

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,5 +1,316 @@
+# Overview
+
+## Technical
```
Comment (Wh1isper) [inline CONTRIBUTING.md:3]: I actually don't think this part is necessary and introduces more maintenance costs...

We're just not writing resumes

Comment (jalr4ever) [reply CONTRIBUTING.md:3]: > I actually don't think this part is necessary and introduces more maintenance costs...
> 
> We're just not writing resumes

All work is being gradually updated in progress. I think the current stage of the SDG needs this, as we need more developers to get involved. Based on the current progress, this part of the maintenance  cost is acceptable. However, it seems that version number information can be removed as it has no significance.

Comment (Wh1isper) [reply CONTRIBUTING.md:3]: Maybe we just need to list a few of them as a reminder of which technologies we are based on? For example, `OpenAI` makes no sense, it's just a client.

Comment (jalr4ever) [reply CONTRIBUTING.md:3]: 🤣 Indeed, let me take a look. Delete some unnecessary ones.

**Comment by jalr4ever** [comment]
@Wh1isper Can you help me review it? If there are no issues, please approve this PR.

**Review by Wh1isper — APPROVED** [review]
LGTM, thanks!

**Pull request merged**